revocation/revocation.go

```diff
@@ -16,26 +16,49 @@
 package revocation
 
 import (
+	"context"
 	"crypto/x509"
 	"errors"
+	"fmt"
 	"net/http"
+	"sync"
 	"time"
 
+	"github.com/notaryproject/notation-core-go/revocation/crl"
 	"github.com/notaryproject/notation-core-go/revocation/ocsp"
 	"github.com/notaryproject/notation-core-go/revocation/result"
+	coreX509 "github.com/notaryproject/notation-core-go/x509"
 )
 
 // Revocation is an interface that specifies methods used for revocation checking
 type Revocation interface {
 	// Validate checks the revocation status for a certificate chain using OCSP
-	// and returns an array of CertRevocationResults that contain the results
-	// and any errors that are encountered during the process
+	// and CRL if OCSP is not available. It returns an array of
+	// CertRevocationResults that contain the results and any errors that are
+	// encountered during the process
 	Validate(certChain []*x509.Certificate, signingTime time.Time) ([]*result.CertRevocationResult, error)
 }
 
+// Options specifies values that are needed to check revocation
+type Options struct {
+	// Ctx is a required context used for the revocation check
+	Ctx context.Context
```

Comment:
Updated.

Reply:
I see `context.Background()` is used in `New` and `NewTimestamp` constructors. They should use the context from caller as well. Instead of adding a `ctx` under `revocation` struct, another option is to follow https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/net/http/request.go;l=346 and https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/net/http/request.go;l=362